api/src/main/java/com/ichi2/anki/api/AddContentApi.java

```diff
@@ -151,6 +158,47 @@ public int addNotes(long modelId, long deckId, List<String[]> fieldsList, List<S
         return getCompat().insertNotes(deckId, newNoteValuesList.toArray(new ContentValues[newNoteValuesList.size()]));
     }
 
+    /**
+     * Add a media file to AnkiDroid's media collection. You would likely supply this uri through a FileProvider, and
+     * then set FLAG_GRANT_READ_URI_PERMISSION using something like:
+     * getContext().grantUriPermission("com.ichi2.anki", uri, Intent.FLAG_GRANT_READ_URI_PERMISSION) and then afterward
+     * remove the permission with getContext().revokePermission(uri, Intent.FLAG_GRAN_READ_URI_PERMISSION)
+     * @param fileUri   Uri for the file to be added.
+     * @param preferredName String that will be added to the beginning of the file name in the media collection
+     * @param mimeType  String indicating the mimeType of the media. Accepts "audio" or "image"
+     * @return the correctly formatted String for the media file to be placed in the desired field of a Card.
+     */
+    public String addMediaFromUri(Uri fileUri, String preferredName, String mimeType) {
+        ContentValues contentValues = new ContentValues();
+        contentValues.put(AnkiMedia.FILE_URI, fileUri.toString());
+        preferredName = preferredName.replace(" ", "_");
+        contentValues.put(AnkiMedia.PREFERRED_NAME, preferredName);
+
+        try {
+            Uri returnUri = mResolver.insert(AnkiMedia.CONTENT_URI, contentValues);
+            // get the filename from Uri, return [sound:%s] % file.getName()
+            String fname = new File(returnUri.getPath()).toString();
+            String formatted_fname = formatMediaName(fname, mimeType);
+            return formatted_fname;
+        } catch (Exception e){
+            return null;
+        }
+
+    }
+
+    public String formatMediaName(String fname, String mimeType) {
```

Comment:
Should protect these from being sent in null as they are accessed blindly, and I think the method can be private so you don't have to worry as much about API contracts as it won't be exposed to other objects. Can return null so should be marked as such as well

```suggestion
    private @Nullable String formatMediaName(@NonNull String fname, @NonNull String mimeType) {
```